src/uu/date/src/date.rs

```diff
@@ -219,19 +219,27 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 let iter = std::iter::once(date);
                 Box::new(iter)
             }
-            DateSource::File(ref path) => match File::open(path) {
-                Ok(file) => {
-                    let lines = BufReader::new(file).lines();
-                    let iter = lines.filter_map(Result::ok).map(parse_date);
-                    Box::new(iter)
-                }
-                Err(_err) => {
+            DateSource::File(ref path) => {
+                if path.is_dir() {
                     return Err(USimpleError::new(
                         2,
-                        format!("{}: No such file or directory", path.display()),
+                        format!("expected file, got directory"),
                     ));
                 }
-            },
+                match File::open(path) {
+                    Ok(file) => {
+                        let lines = BufReader::new(file).lines();
+                        let iter = lines.map_while(Result::ok).map(parse_date);
+                        Box::new(iter)
+                    }
+                    Err(_err) => {
+                        return Err(USimpleError::new(
+                            2,
+                            String::new()
+                        ));
+                    }
+                }
```

Comment:
```suggestion
                let file = File::open(path).map_err_context(|| path.quote())?;
                let lines = BufReader::new(file).lines();
                let iter = lines.map_while(Result::ok).map(parse_date);
                Box::new(iter)
```

We can use `map_err_context` to map an `io::Result` to a `UResult` while also giving an informative error message. You might need to import this trait: https://docs.rs/uucore/0.0.17/uucore/error/trait.FromIo.html

If I did this correctly the error message should be the same as before this change, but not always "No such file or directory".

Reply:
I had to add `to_string()` for `path.quote()` but it works. I have also updated unit test.